src/Silex/Provider/VarDumperServiceProvider.php

```diff
@@ -45,13 +50,14 @@ public function boot(Application $app)
             return;
         }
 
-        // This code is here to lazy load the dump stack. This default
-        // configuration for CLI mode is overridden in HTTP mode on
-        // 'kernel.request' event
+        // This code is here to lazy load the dump stack.
         VarDumper::setHandler(function ($var) use ($app) {
-            VarDumper::setHandler($handler = function ($var) use ($app) {
-                $app['var_dumper.cli_dumper']->dump($app['var_dumper.cloner']->cloneVar($var));
-            });
+            $handler = function ($var) use ($app) {
+                'cli' === PHP_SAPI
```

Comment:
This could be configured with a default value in `register` method.

```php
function register(Container $app) {
    ...
    $app['var_dumper.is_cli'] = PHP_SAPI;
    ...
}
```